include/boost/url/url_base.hpp

```diff
@@ -1615,17 +1615,32 @@ class url_base
 
 //----------------------------------------------------------
 
-/** A read-only view to the path segments.
+/** A read/write view to the path segments.
 */
 class url_base::segments_type
 {
     url_base* v_ = nullptr;
 
 public:
+
     class value_type;
     class iterator;
     using const_iterator = iterator;
 
+private:
+
+    // Strong guarantee.
```

Comment:
For private members I put these types of comments on the definition rather than the declaration. A minor point.